presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -7187,4 +7188,18 @@ public void testRewriteContainsToIn()
         assertQuery(session, "select x, contains(array[], x) from (values 1, 2, 3, 4, null) t(x)", "values (1, false), (2, false), (3, false), (4, false), (null, null)");
         assertQuery(session, "select x, contains(cast(null as array<bigint>), x) from (values 1, 2, 3, 4, null) t(x)", "values (1, null), (2, null), (3, null), (4, null), (null, null)");
     }
+
+    @Test
+    public void testRemoveRedundantCastToVarcharInJoinClause()
+    {
+        Session session = Session.builder(getSession())
+                .setSystemProperty(REMOVE_REDUNDANT_CAST_TO_VARCHAR_IN_JOIN, "true")
+                .build();
+        // Trigger optimization
+        assertQuery(session, "select * from orders o join customer c on cast(o.custkey as varchar) = cast(c.custkey as varchar)");
```

Comment:
can you add a case where there is a cross join on top of the join you're optimizing? I think your optimization covers it, but cross join can be sensitive if new columns are introduced or reordered in one of the children, so it's good to add a test to make sure there are no regressions
something like this:
```
select * from 
(select * from orders o join customer c on cast(o.custkey as varchar) = cast(c.custkey as varchar)) r, nation;
```